hal/sparcv8leon3/arch/cpu.h

```diff
@@ -233,12 +224,15 @@ static inline unsigned int hal_cpuGetCount(void)
 
 static inline unsigned int hal_cpuGetID(void)
 {
-	/* TODO: doesn't work for some reason */
-	// u32 asr17;
+	u32 asr17 = 0;
```

Comment:
Hmm, this var init shouldn't be needed
```suggestion
	u32 asr17;
```